diff --git a/lualib-src/lua-socket.c b/lualib-src/lua-socket.c
index b8c5cc7..b21261e 100644
--- a/lualib-src/lua-socket.c
+++ b/lualib-src/lua-socket.c
@@ -11,11 +11,16 @@
 #include <lua.h>
 #include <lauxlib.h>
 
+#ifdef _WIN32
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <netinet/in.h>
+#endif
 
 #include "skynet.h"
 #include "skynet_socket.h"
@@ -468,6 +473,6 @@ lconnect(lua_State *L) {
 	size_t sz = 0;
 	const char * addr = luaL_checklstring(L,1,&sz);
-	char tmp[sz];
+	char *tmp = lua_newuserdatauv(L, sz == 0 ? 1 : sz, 0);
 	int port = 0;
 	const char * host = address_port(L, tmp, addr, 2, &port);
 	if (port == 0) {
@@ -501,6 +506,6 @@ llisten(lua_State *L) {
 	size_t sz = 0;
 	const char * addr = luaL_checklstring(L, 1, &sz);
-	char tmp[sz];
+	char *tmp = lua_newuserdatauv(L, sz == 0 ? 1 : sz, 0);
 	int port = 0;
 	const char * host = address_port(L, tmp, addr, 2, &port);
 	int backlog = luaL_optinteger(L,3,BACKLOG);
@@ -658,6 +663,6 @@ ludp(lua_State *L) {
 	size_t sz = 0;
 	const char * addr = lua_tolstring(L,1,&sz);
-	char tmp[sz];
+	char *tmp = lua_newuserdatauv(L, sz == 0 ? 1 : sz, 0);
 	int port = 0;
 	const char * host = NULL;
 	if (addr) {
@@ -679,6 +684,6 @@ ludp_connect(lua_State *L) {
 	size_t sz = 0;
 	const char * addr = luaL_checklstring(L,2,&sz);
-	char tmp[sz];
+	char *tmp = lua_newuserdatauv(L, sz == 0 ? 1 : sz, 0);
 	int port = 0;
 	const char * host = NULL;
 	if (addr) {
@@ -698,6 +703,6 @@ ludp_dial(lua_State *L) {
 	size_t sz = 0;
 	const char * addr = luaL_checklstring(L, 1, &sz);
-	char tmp[sz];
+	char *tmp = lua_newuserdatauv(L, sz == 0 ? 1 : sz, 0);
 	int port = 0;
 	const char * host =  address_port(L, tmp, addr, 2, &port);
 
@@ -716,6 +721,6 @@ ludp_listen(lua_State *L) {
 	size_t sz = 0;
 	const char * addr = luaL_checklstring(L, 1, &sz);
-	char tmp[sz];
+	char *tmp = lua_newuserdatauv(L, sz == 0 ? 1 : sz, 0);
 
 	int port = 0;
 	const char * host = address_port(L, tmp, addr, 2, &port);
